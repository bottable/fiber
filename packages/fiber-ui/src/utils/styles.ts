import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
