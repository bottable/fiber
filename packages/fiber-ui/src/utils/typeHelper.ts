export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
