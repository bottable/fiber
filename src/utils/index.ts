export * from './typeHelper'
export * from './mergeProps'
