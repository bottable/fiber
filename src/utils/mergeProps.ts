import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
