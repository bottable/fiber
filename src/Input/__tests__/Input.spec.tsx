import { Input } from '..'

import React from 'react'
import { render } from 'test/utils'
import { MdSettings } from 'react-icons/md'

describe('Input', () => {
  it('renders correctly', () => {
    const block = render(
      <>
        <Input />
        <div>
          <Input size='lg' placeholder='Large Input' />
          <br />
          <br />
          <Input placeholder='Medium Input' />
          <br />
          <br />
          <Input size='sm' placeholder='Small Input' />
        </div>
        <div>
          <Input addonBefore='http://' addonAfter='.com' />
          <br />
          <br />
          <Input addonAfter={<MdSettings />} />
        </div>
        <div>
          <Input prefix='￥' suffix='RMB' />
          <br />
          <br />
          <Input prefix='￥' suffix='RMB' disabled />
          <br />
          <br />
          <Input addonBefore='http://' suffix='.com' />
        </div>

        <div>
          <Input placeholder='Borderless' bordered={false} />
        </div>

        <div>
          <Input placeholder='Disabled' disabled />
        </div>
      </>
    )

    expect(block).toMatchSnapshot()
  })
})
