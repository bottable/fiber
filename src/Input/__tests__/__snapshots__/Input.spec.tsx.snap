// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
Array [
  <input
    className="sc-AxirZ sc-AxhCb blVfAb"
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />,
  <div>
    <input
      className="sc-AxirZ sc-AxhUy kMpcSi"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Large Input"
      value=""
    />
    <br />
    <br />
    <input
      className="sc-AxirZ sc-AxhCb blVfAb"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Medium Input"
      value=""
    />
    <br />
    <br />
    <input
      className="sc-AxirZ sc-AxiKw fjNudk"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Small Input"
      value=""
    />
  </div>,
  <div>
    <span
      className="sc-Axmtr dxDwIG"
    >
      <span
        className="sc-AxheI eElXNJ"
      >
        <span
          className="sc-AxgMl kjSpuB"
        >
          http://
        </span>
        <input
          className="sc-AxirZ sc-AxhCb jQilkc"
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
        <span
          className="sc-AxgMl kjSpuB"
        >
          .com
        </span>
      </span>
    </span>
    <br />
    <br />
    <span
      className="sc-Axmtr dxDwIG"
    >
      <span
        className="sc-AxheI eElXNJ"
      >
        <input
          className="sc-AxirZ sc-AxhCb dztiSk"
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
        <span
          className="sc-AxgMl kjSpuB"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
              transform="scale(1.2, 1.2)"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>,
  <div>
    <span
      className="sc-AxmLO ljvmGI"
      onClick={[Function]}
    >
      ￥
      <input
        className="sc-AxirZ sc-AxhCb brbPJA"
        onChange={[Function]}
        onKeyDown={[Function]}
        value=""
      />
      RMB
    </span>
    <br />
    <br />
    <span
      className="sc-AxmLO gUVxHk"
      disabled={true}
      onClick={[Function]}
    >
      ￥
      <input
        className="sc-AxirZ sc-AxhCb coWndV"
        disabled={true}
        onChange={[Function]}
        onKeyDown={[Function]}
        value=""
      />
      RMB
    </span>
    <br />
    <br />
    <span
      className="sc-Axmtr dxDwIG"
    >
      <span
        className="sc-AxheI eElXNJ"
      >
        <span
          className="sc-AxgMl kjSpuB"
        >
          http://
        </span>
        <span
          className="sc-AxmLO jUTKb"
          onClick={[Function]}
        >
          <input
            className="sc-AxirZ sc-AxhCb brbPJA"
            onChange={[Function]}
            onKeyDown={[Function]}
            value=""
          />
          .com
        </span>
      </span>
    </span>
  </div>,
  <div>
    <input
      className="sc-AxirZ sc-AxhCb hfhPdo"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Borderless"
      value=""
    />
  </div>,
  <div>
    <input
      className="sc-AxirZ sc-AxhCb jjecpj"
      disabled={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Disabled"
      value=""
    />
  </div>,
]
`;
